src/algebraic_topology/fundamental_groupoid/simply_connected.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+import algebraic_topology.fundamental_groupoid.induced_maps
+import topology.homotopy.contractible
+import category_theory.punit
+import algebraic_topology.fundamental_groupoid.punit
+
+/-!
+# Simply connected spaces
+This file defines simply connected spaces.
+A topological space is simply connected if its fundamental groupoid is equivalent to `unit`.
+
+## Main theorems
+  - `simply_connected_iff_unique_homotopic` - A space is simply connected if and only if it is
+    nonempty and there is a unique path up to homotopy between any two points
+
+  - `simply_connected_space.of_contractible` - A contractible space is simply connected
+-/
+noncomputable theory
+
+open category_theory
+open continuous_map
+open_locale continuous_map
+
+/-- A simply connected space is one whose fundamental groupoid is equivalent to `discrete unit` -/
+class simply_connected_space (X : Type*) [topological_space X] : Prop :=
+(equiv_unit [] : nonempty (fundamental_groupoid X ≌ discrete unit))
+
+lemma simply_connected_def (X : Type*) [topological_space X] :
+  simply_connected_space X ↔ nonempty (fundamental_groupoid X ≌ discrete unit) :=
+⟨λ h, @simply_connected_space.equiv_unit X _ h, λ h, ⟨h⟩⟩
+
+lemma simply_connected_iff_unique_homotopic (X : Type*) [topological_space X] :
+  simply_connected_space X ↔ (nonempty X) ∧
+  ∀ (x y : X), nonempty (unique (path.homotopic.quotient x y)) :=
+by { rw [simply_connected_def, equiv_punit_iff_unique], refl, }
+
+namespace simply_connected_space
+variables {X : Type*} [topological_space X] [simply_connected_space X]
+
+instance (x y : X) : subsingleton (path.homotopic.quotient x y) :=
+@unique.subsingleton _ (nonempty.some (by { rw simply_connected_iff_unique_homotopic at *, tauto }))
```

Comment:
`unique.subsingleton` is an instance so you may as well just give the stronger `unique` instance (or add a `nonempty` instance). `unique (fundamental_group X x)` can also be obtained directly from that, but you probably want `to_inhabited.default` to be defeq to the class of the constant path.